src/route/dataset.ts

```diff
@@ -252,6 +257,238 @@ router.get('/:dataset_id/view', loadDataset(), async (req: Request, res: Respons
     res.json(processedCSV);
 });
 
+// GET /dataset/:dataset_id/cube
+// Returns the latest revision of the dataset as a DuckDB File
+router.get('/:dataset_id/cube', loadDataset(), async (req: Request, res: Response, next: NextFunction) => {
+    const dataset = res.locals.dataset;
+    const lang = req.language.split('-')[0];
+    const latestRevision = getLatestRevision(dataset);
+    if (!latestRevision) {
+        next(new UnknownException('errors.no_revision'));
+        return;
+    }
+    let cubeFile: string;
+    if (latestRevision.onlineCubeFilename) {
+        const dataLakeService = new DataLakeService();
+        const fileBuffer = await dataLakeService.getFileBuffer(latestRevision.onlineCubeFilename, dataset.id);
+        cubeFile = tmp.tmpNameSync({ postfix: '.duckdb' });
+        fs.writeFileSync(cubeFile, fileBuffer);
+    } else {
+        try {
+            cubeFile = await createBaseCube(dataset, latestRevision);
```

Comment:
Thats my longer term plan...  But the earliest preview requests probably don't have any form of generated cube.  In these case we will need to have the option to generate here and just send the result to the user.  The most basic cube just has a fact table in it and the generating process takes fractions of a second because none of the more complex queries need to be done.